runtime-modules/content/src/migrations.rs

```diff
@@ -0,0 +1,219 @@
+                                        .insert(ContentModerationAction::DeleteVideoAssets(*b));
+                                }
+                                ContentModerationActionV0::DeleteNonVideoChannelAssets => {
+                                    permissions_v1.insert(
+                                        ContentModerationAction::DeleteNonVideoChannelAssets,
+                                    );
+                                }
+                                ContentModerationActionV0::UpdateChannelNftLimits => {
+                                    permissions_v1
+                                        .insert(ContentModerationAction::UpdateChannelNftLimits);
+                                }
+                                ContentModerationActionV0::DeleteVideo => {}
+                                ContentModerationActionV0::DeleteChannel => {}
+                            });
+                            permissions_by_level.insert(*level, permissions_v1);
+                        });
+                    let mut group_created_successfully = false;
+                    let mut group = CuratorGroupRecord::try_create::<T>(
+                        old_value.active,
+                        &permissions_by_level,
+                    )
+                    .map_or_else(
+                        |_err| Default::default(),
+                        |v| {
+                            group_created_successfully = true;
+                            v
+                        },
+                    );
+                    if !group_created_successfully {
+                        // The group will be dropped
+                        log!(warn, "Failed to create a group");
+                        return None;
+                    }
+                    old_value.curators.into_iter().for_each(|(k, v)| {
+                        let result = group.try_add_curator::<T>(k, &v);
+                        if result.is_err() {
+                            log!(warn, "Failed to add curator to group");
+                        }
+                    });
+                    Some(group)
+                },
+            );
+
+            current.put::<Pallet<T>>();
+
+            <T as frame_system::Config>::BlockWeights::get().max_block
+        }
+
+        #[cfg(feature = "try-runtime")]
+        fn post_upgrade(state: Vec<u8>) -> Result<(), &'static str> {
+            let onchain = Pallet::<T>::on_chain_storage_version();
+            ensure!(onchain < 2, "this migration needs to be removed");
+            ensure!(onchain == 1, "this migration needs to be run");
+            let group_count = CuratorGroupById::<T>::iter().count();
+            log!(
+                info,
+                "Number of curator groups post_upgrade {}",
+                group_count
+            );
+            let pre_upgrade_group_count = usize::from_be_bytes(state.try_into().unwrap());
+            ensure!(
```

Comment:
I'm not sure I understand the consequences of this sort of check at runtime, is that really appropriate? I see the case for asserting this in debug mode or something, but if there is some sort of bug, is there any added profit from failing this hook? what happens then?

Reply:
When we build the production runtime we will not build it with the `try-runtime` feature, so the pre_/post_ methods will not exist and will not execute.

The feature is meant specifically for testing a new runtime upgrade during development against some existing chain state.